ee/server/service/users.go

```diff
@@ -0,0 +1,38 @@
+package service
+
+import (
+	"context"
+	"errors"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+)
+
+// GetSSOUser is the premium implementation of svc.GetSSOUser, it allows to
+// create users during the SSO flow the first time they log in if
+// config.SSOSettings.EnableJITProvisioning is `true`
+func (svc *Service) GetSSOUser(ctx context.Context, auth fleet.Auth) (*fleet.User, error) {
+	config, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting app config")
+	}
+
+	user, err := svc.Service.GetSSOUser(ctx, auth)
+	var nfe fleet.NotFoundError
+	if !errors.As(err, &nfe) || !config.SSOSettings.EnableJITProvisioning {
+		return user, err
+	}
```

Comment:
Nit:
```go
user, err := svc.Service.GetSSOUser(ctx, auth)
var nfe fleet.NotFoundError
switch {
case err == nil:
  // OK
case errors.As(err, &nfe):
  if !config.SSOSettings.EnableJITProvisioning {
    return nil, err
  }
default:
  return nil, err
}
```
I like the idiom of explicitly returning zero types (`nil` for `*fleet.User` in this case) when there's an error (`err != nil`). (Otherwise it's not clear to me why the developer is trying to possibly return a `user` even if `err != nil`.)

Reply:
thanks! I very much like the explicitness of this, will update